04/2_print_string_buffer/buffered_print.c

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "buffered_print.h"
+
+#define BUFFER_LENGTH (32)
+
+static int s_buffer_next_index = 0;
+static char s_buffer[BUFFER_LENGTH];
+
+void buffered_print(const char* src, size_t num)
+{
+    if (s_buffer_next_index == BUFFER_LENGTH) {

```

Comment:
this doesn't work.. when thess strings are passed into the function (if buffer length was 16)

 "12345678901234"
"51234567890123456"

this should print 

"123456789012345"
"123456789012345"